Assignment4/vcuende-180188/task07.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1Oxpxs_IQi6xHcWEmRl-eJOSzUUKWGni5
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+NS = Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    ?Subject rdfs:subClassOf ns:Person. 
+  }
+  ''',
+  initNs = { "ns": NS}
+)
+
+for r in g.query(q1):
+  print(r.Subject)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+NS = Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    {?Subject rdf:type ns:Person}
+    UNION
+    {?Subject rdf:type ns:Researcher}
```

Comment:
This is not correct, even if it works with the data at hand. The query should retrieve all possible subclasses of person, not just researcher.